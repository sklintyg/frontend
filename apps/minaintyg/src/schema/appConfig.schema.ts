import { z } from 'zod'

export const appConfigLinkSchema = z.object({
  key: z.string(),
  url: z.string(),
  text: z.string(),
  target: z.string(),
})

export const appConfigSchema = z.object({
  useMinifiedJavascript: z.boolean(),
  version: z.string(),
  buildNumber: z.string(),
  elva77MainUrl: z.string(),
  elva77LoginUrl: z.string(),
  applicationLogoutUrl: z.string(),
  miUserSurveyUrl: z.string(),
  miUserSurveyVersion: z.string(),
  miUserSurveyDateFrom: z.string(),
  miUserSurveyDateTo: z.string(),
  knownRecipients: z.array(z.object({ id: z.string(), name: z.string(), trusted: z.boolean() })),
  links: z.record(appConfigLinkSchema),
  banners: z.array(z.unknown()),
})

export type AppConfig = z.infer<typeof appConfigSchema>
export type AppConfigLink = z.infer<typeof appConfigLinkSchema>
