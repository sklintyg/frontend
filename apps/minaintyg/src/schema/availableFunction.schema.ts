import { z } from 'zod'

export const AvailableFunctionType = z.enum(['SEND_CERTIFICATE'])
export const AvailableFunctionInformationType = z.enum([''])

export const availableFunctionInformationSchema = z.object({
  type: AvailableFunctionInformationType,
  id: z.string(),
})

export const availableFunctionSchema = z.object({
  body: z.string().nullable(),
  description: z.string().nullable(),
  name: z.string().nullable(),
  title: z.string().nullable(),
  type: AvailableFunctionType,
  information: z.array(availableFunctionInformationSchema),
})

export type AvailableFunction = z.infer<typeof availableFunctionSchema>
