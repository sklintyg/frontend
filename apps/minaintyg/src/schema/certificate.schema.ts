import { z } from 'zod'

export const CertificateStatusEnum = z.enum(['NEW', 'REPLACED', 'SENT', 'NOT_SENT'])

export const certificateEventSchema = z.object({
  timestamp: z.string().datetime(),
  certificateId: z.string(),
  description: z.string(),
})

export const certificateSummarySchema = z.object({ label: z.string(), value: z.string() }).optional()

export const certificateTypeSchema = z.object({
  id: z.string(),
  version: z.string(),
  description: z.string().optional(),
})

export const certificateIssuerSchema = z.object({
  name: z.string(),
  phoneNumber: z.string(),
})

export const certificateUnitSchema = z.object({
  id: z.string(),
  name: z.string(),
  address: z.string(),
})

export const certificateMetadataSchema = z.object({
  issuer: certificateIssuerSchema,
  unit: certificateUnitSchema,
  name: z.string(),
  events: z.array(certificateEventSchema),
  statuses: z.array(CertificateStatusEnum),
  id: z.string(),
  issued: z.string().datetime(),
  type: certificateTypeSchema,
  summary: certificateSummarySchema,
})

export const certificateContentSchema = z.object({ heading: z.string(), body: z.string() })

export const certificateSchema = z.object({
  metadata: certificateMetadataSchema,
  content: z.array(certificateContentSchema),
})

export type CertificateContent = z.infer<typeof certificateContentSchema>
export type CertificateEvent = z.infer<typeof certificateEventSchema>
export type CertificateIssuer = z.infer<typeof certificateIssuerSchema>
export type CertificateMetadata = z.infer<typeof certificateMetadataSchema>
export type CertificateStatus = z.infer<typeof CertificateStatusEnum>
export type CertificateSummary = z.infer<typeof certificateSummarySchema>
export type CertificateUnit = z.infer<typeof certificateUnitSchema>
export type Certificate = z.infer<typeof certificateSchema>
