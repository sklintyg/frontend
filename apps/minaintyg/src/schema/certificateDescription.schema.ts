import { z } from 'zod'

export const certificateDescriptionSchema = z.object({
  id: z.string(),
  label: z.string(),
  description: z.string(),
  detailedDescription: z.string(),
  issuerTypeId: z.string(),
  cssPath: z.string(),
  scriptPath: z.string(),
  dependencyDefinitionPath: z.string(),
  defaultRecipient: z.string(),
  deprecated: z.boolean(),
  displayDeprecated: z.boolean(),
})

export type CertificateDescription = z.infer<typeof certificateDescriptionSchema>
