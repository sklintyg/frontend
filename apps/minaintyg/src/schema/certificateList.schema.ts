import { z } from 'zod'

export enum CertificateStatus {
  NEW = 'Nytt',
  REPLACED = 'Ers√§tter intyg',
  SENT = 'Skickat',
  NOT_SENT = 'Ej skickat',
}

export const certificateListSummarySchema = z.object({ label: z.string(), value: z.string() }).optional()

export const certificateListEventSchema = z.object({
  timestamp: z.string().datetime(),
  certificateId: z.string(),
  description: z.string(),
})

export const certificateListIssuerSchema = z.object({ name: z.string() })
export const certificateUnitSchema = z.object({ id: z.string(), name: z.string() })
export const certificateTypeSchema = z.object({
  id: z.string(),
  name: z.string(),
  version: z.string(),
})

export const certificateListItemSchema = z.object({
  summary: certificateListSummarySchema,
  issuer: certificateListIssuerSchema,
  unit: certificateUnitSchema,
  events: z.array(certificateListEventSchema),
  statuses: z.array(z.nativeEnum(CertificateStatus)),
  id: z.string(),
  issued: z.string().datetime(),
  type: certificateTypeSchema,
})

export type CertificateListEvent = z.infer<typeof certificateListEventSchema>
export type CertificateListIssuer = z.infer<typeof certificateListIssuerSchema>
export type CertificateListUnit = z.infer<typeof certificateUnitSchema>
export type CertificateListSummary = z.infer<typeof certificateListSummarySchema>
export type CertificateListItem = z.infer<typeof certificateListItemSchema>
