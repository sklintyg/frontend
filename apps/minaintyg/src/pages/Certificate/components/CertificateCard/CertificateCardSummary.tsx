import { useFormat } from '../../../../hooks/useFormat'
import { CertificateSummary } from '../../../../schema/certificate.schema'

export function CertificateCardSummary({ summary, timestamp }: { summary: CertificateSummary; timestamp: string }) {
  const { date } = useFormat()

  return (
    <div className="mb-2.5 flex flex-col place-content-end gap-2.5 border-stone-line md:mb-5 md:flex-row md:border-b md:pb-5">
      {summary && (
        <div className="grow leading-5">
          <h4 className="ids-heading-4 mb-0 leading-5 md:inline-block md:after:content-[':']">{summary.label}</h4> {summary.value}
        </div>
      )}
      <div className="flex flex-col leading-5 md:flex-row md:gap-1.5 whitespace-nowrap">
        <h4 className="ids-heading-4 mb-0 leading-5 md:inline-block md:after:content-[':']">Intyg utf√§rdat</h4>
        <span>{date(timestamp)}</span>
      </div>
    </div>
  )
}
