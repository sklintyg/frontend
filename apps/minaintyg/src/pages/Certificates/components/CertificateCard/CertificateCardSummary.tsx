import { useFormat } from '../../../../hooks/useFormat'
import { CertificateListSummary } from '../../../../schema/certificateList.schema'

export function CertificateCardSummary({ summary, timestamp }: { summary: CertificateListSummary; timestamp: string }) {
  const { formatDate } = useFormat()

  return (
    <div className="mb-2.5 flex flex-col place-content-end gap-2.5 md:mb-5 md:flex-row md:border-b md:pb-5">
      {summary.length > 0 && (
        <p className="grow">
          <span className="font-bold">Avser </span>
          {summary.map(([description, value]) => (
            <>
              <span className="font-bold">{description}:</span> {value}{' '}
            </>
          ))}
        </p>
      )}
      <div className="flex flex-col md:flex-row md:gap-1.5">
        <span className="font-bold">Intyg utf√§rdat:</span>
        <span>{formatDate(timestamp)}</span>
      </div>
    </div>
  )
}
