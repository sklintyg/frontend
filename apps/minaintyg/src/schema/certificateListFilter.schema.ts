import { z } from 'zod'
import { CertificateStatus } from './certificateList.schema'

export const certificateFilterOptionsSchema = z.object({
  total: z.number(),
  statuses: z.array(z.nativeEnum(CertificateStatus)),
  units: z.array(z.string()),
  certificateTypes: z.array(z.string()),
  years: z.array(z.string()),
})

type SelectedOptions<T> = { [X in keyof T]: T[X] extends Array<unknown> ? T[X][number] : T[X] }

export type CertificateFilterOptions = z.infer<typeof certificateFilterOptionsSchema>
export type CertificateSelectedOptions = SelectedOptions<CertificateFilterOptions>
