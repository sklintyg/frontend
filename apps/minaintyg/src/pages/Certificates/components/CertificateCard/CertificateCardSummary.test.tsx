import { render } from '@testing-library/react'
import { CertificateListSummary } from '../../../../schema/certificateList.schema'
import { CertificateCardSummary } from './CertificateCardSummary'

it('Should render correctly with summary', () => {
  const summary: CertificateListSummary = [
    ['intygsperiod', '2023-05-09 - 2023-06-23'],
    ['diagnos', 'Downs syndrom'],
  ]
  const { baseElement } = render(<CertificateCardSummary summary={summary} timestamp="2023-05-09T18:37:23.159Z" />)
  expect(baseElement).toMatchSnapshot()
})

it('Should render correctly without summary', () => {
  const { baseElement } = render(<CertificateCardSummary summary={[]} timestamp="2023-05-09T18:37:23.159Z" />)
  expect(baseElement).toMatchSnapshot()
})
