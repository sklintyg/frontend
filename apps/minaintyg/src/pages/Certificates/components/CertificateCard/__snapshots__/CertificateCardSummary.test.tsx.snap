// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render correctly with summary 1`] = `
<body>
  <div>
    <div
      class="mb-2.5 flex flex-col place-content-end gap-2.5 md:mb-5 md:flex-row md:border-b md:pb-5"
    >
      <p
        class="grow"
      >
        <span
          class="font-bold"
        >
          Avser 
        </span>
        <span>
          <span
            class="font-bold"
          >
            intygsperiod
            :
          </span>
           
          2023-05-09 - 2023-06-23
           
        </span>
        <span>
          <span>
            & 
          </span>
          <span
            class="font-bold"
          >
            diagnos
            :
          </span>
           
          Downs syndrom
           
        </span>
      </p>
      <div
        class="flex flex-col md:flex-row md:gap-1.5"
      >
        <span
          class="font-bold"
        >
          Intyg utfärdat:
        </span>
        <span>
          2023-05-09
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`Should render correctly without summary 1`] = `
<body>
  <div>
    <div
      class="mb-2.5 flex flex-col place-content-end gap-2.5 md:mb-5 md:flex-row md:border-b md:pb-5"
    >
      <div
        class="flex flex-col md:flex-row md:gap-1.5"
      >
        <span
          class="font-bold"
        >
          Intyg utfärdat:
        </span>
        <span>
          2023-05-09
        </span>
      </div>
    </div>
  </div>
</body>
`;
